src/Core/Prefab/Globe/Atmosphere.js

```diff
@@ -15,186 +17,219 @@ import groundVS from './Shaders/groundVS.glsl';
-        fInnerRadius: { value: atmosphere.innerRadius },
-        fInnerRadius2: { value: atmosphere.innerRadius * atmosphere.innerRadius },
-        fOuterRadius: { value: atmosphere.outerRadius },
-        fOuterRadius2: { value: atmosphere.outerRadius * atmosphere.outerRadius },
-        fKrESun: { value: atmosphere.Kr * atmosphere.ESun },
-        fKmESun: { value: atmosphere.Km * atmosphere.ESun },
-        fKr4PI: { value: atmosphere.Kr * 4.0 * Math.PI },
-        fKm4PI: { value: atmosphere.Km * 4.0 * Math.PI },
-        fScale: { value: 1 / (atmosphere.outerRadius - atmosphere.innerRadius) },
-        fScaleDepth: { value: atmosphere.scaleDepth },
-        fScaleOverScaleDepth: { value: 1 / (atmosphere.outerRadius - atmosphere.innerRadius) / atmosphere.scaleDepth },
-        g: { value: atmosphere.g },
-        g2: { value: atmosphere.g * atmosphere.g },
-        nSamples: { value: 3 },
-        fSamples: { value: 3.0 },
-        tDisplacement: { value: new THREE.Texture() },
-        tSkyboxDiffuse: { value: new THREE.Texture() },
-        fNightScale: { value: 1.0 },
-    };
-
-    this.ground = {
-        geometry: new THREE.SphereGeometry(atmosphere.innerRadius, 50, 50),
-        material: new THREE.ShaderMaterial({
+    update(context, layer, node) {
+        // update uniforms
+        node.material.fogDistance = this.fog.distance;
+        node.material.lightingEnabled = this.realisticAtmosphere.visible;
+        node.material.lightPosition = this.realisticLightingPosition;
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    preUpdate(context, srcs) {
+        const cameraPosition = context.view.camera.camera3D.position;
+        if (this.fog.enable) {
+            v.setFromMatrixPosition(context.view.tileLayer.object3d.matrixWorld);
+            const len = v.distanceTo(cameraPosition);
+            // Compute fog distance, this function makes it possible to have a shorter distance
+            // when the camera approaches the ground
+            this.fog.distance = mfogDistance * ((len - ellipsoidSizes.x * 0.99) * 0.25 / ellipsoidSizes.x) ** 1.5;
+        } else {
+            this.fog.distance = 10e10;
+        }
+
+        const renderer = context.view.mainLoop.gfxEngine.renderer;
+        // get altitude camera
+        coordCam.set(context.view.referenceCrs, cameraPosition).as('EPSG:4326', coordGeoCam);
+        const altitude = coordGeoCam.altitude();
+
+        // If the camera altitude is below limitAlti,
+        // we interpolate between the sky color and the space color
+        if (altitude < limitAlti) {
+            const t = (limitAlti - altitude) / limitAlti;
+            colorSky.copy(spaceColor).lerp(skyBaseColor, t);
+            renderer.setClearColor(colorSky, renderer.getClearAlpha());
+        } else {
+            renderer.setClearColor(spaceColor, renderer.getClearAlpha());
+        }
+    }
+
+    // default to non-realistic lightning
+    _initRealisticLighning() {
```

Comment:
Do we have a source, a link to explain those values ?

Reply:
[`SkyShader.js`](https://github.com/iTowns/itowns/blob/master/src/Core/Prefab/Globe/SkyShader.js) there is link to `skyDome`.

I add this link for [Atmosphere Shader From Space (Atmospheric scattering) ](http://stainlessbeer.weebly.com/planets-9-atmospheric-scattering.html)